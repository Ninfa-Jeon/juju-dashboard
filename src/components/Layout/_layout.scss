$color-sidebar: #003b4e;

.l-container {
  display: grid;
  grid-template-columns: 250px 1fr;
  min-height: 100vh;
}

.l-main {
  overflow: hidden;
  position: relative;
}

.l-side {
  background: $color-sidebar;
  padding: 0;
}
