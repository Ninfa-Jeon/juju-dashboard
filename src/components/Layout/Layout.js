import React from "react";
import PrimaryNav from "components/PrimaryNav/PrimaryNav";

import "./_layout.scss";

const Layout = ({ children }) => {
  return (
    <>
      <div className="l-side">
        <PrimaryNav />
      </div>
      <div className="l-main">
        <main id="main-content">{children}</main>
      </div>
    </>
  );
};

export default Layout;
