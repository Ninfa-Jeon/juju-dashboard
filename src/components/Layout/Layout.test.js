import React from "react";
import { shallow } from "enzyme";

import Layout from "./Layout";

describe("Layout", () => {
  it("renders without crashing and matches snapshot", () => {
    const wrapper = shallow(<Layout />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders with a sidebar", () => {
    const wrapper = shallow(<Layout />);
    expect(wrapper.find(".l-side")).toHaveLength(1);
  });

  it("should display the children", () => {
    const wrapper = shallow(<Layout>content</Layout>);
    expect(wrapper.find("#main-content").html()).toStrictEqual(
      `<main id="main-content">content</main>`
    );
  });
});
