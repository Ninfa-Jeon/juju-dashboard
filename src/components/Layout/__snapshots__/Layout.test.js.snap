// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders without crashing and matches snapshot 1`] = `
<Fragment>
  <div
    className="l-side"
  >
    <PrimaryNav />
  </div>
  <div
    className="l-main"
  >
    <main
      id="main-content"
    />
  </div>
</Fragment>
`;
